lineax/_solver/Woodbury.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any
+from typing_extensions import TypeAlias
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array, PyTree
+
+from .._operator import (
+    AbstractLinearOperator,
+    is_Woodbury,
+    MatrixLinearOperator,
+    woodbury,
+)
+from .._solution import RESULTS
+from .._solve import AbstractLinearSolver, AutoLinearSolver
+from .misc import (
+    pack_structures,
+    PackedStructures,
+    ravel_vector,
+    transpose_packed_structures,
+    unravel_solution,
+)
+
+
+_Woodbury_State: TypeAlias = tuple[
+    tuple[Array, Array, Array],
+    tuple[AbstractLinearSolver, Any, AbstractLinearSolver, Any],
+    PackedStructures,
+]
+
+
+def compute_pushthrough(A_solver, A_state, C, U, V):
```

Comment:
Meanwhile this would have a leading underscore to denote the fact that it is private to this file :)

(Also, can you add type annotations here?)